### ISSUE TITLE
Gruppierte Benchmarks

### ISSUE THREAD

**Issue by fuchsst** [buchen/portfolio#286]
Einzelne Wertpapiere lassen sich im Performance-Diagramm als Benchmark hinzufügen. Schön wäre es wenn es möglich wäre auch die "Aggregation" entsprechend der Klassifizierungen hinzufügen zu können (Quasi analog "Datenreihe hinzufügen...", wo diese mit einem Ordnersymbol aufgelistet werden).

Aufgrund der einerseits voneinander unabhängigen und andererseits z.T. recht erheblichen unterschiedlichen Kursniveaus müsste man dazu zunächst für jedes in der darzustellenden Kategorie enthaltenen Wertpapier die Performancezeitreihe errechnen. Die Gruppenperformance wäre dann der gewichtete Durchschnitt der Einzel-Performances (Gewichtung wie in der Klassifizierung hinterlegt).

**Comment by buchen** [comment]
Für die Klassifikationen werden doch die gruppierten Werte schon aufgeführt. Wenn man in der Asset Allocation "risikobehaftete Werte" und darunter "Small Caps" und "Large Caps" anlegt, dann kann man alle drei auswählen.

**Comment by fuchsst** [comment]
ich meine unter Berichte -> Performance -> Diagramm
Was funktioniert, ist über das "Zahnrad" -> "Datenreihe hinzufügen...", sowohl Einzelwerte, als auch deren Zusammenfassung auf Basis der Klassifizierung in das Diagramm einzufügen.
Was nicht geht (bzw. ich nicht finde) ist analog dazu mit Hilfe des "Zahnrad" -> "Benchmark hinzufügen..." eine Zeitreihe für die Gruppierung (z.B. "Small Caps") hinzuzufügen.

Wenn ich die Benchmarkberechnung richtig verstanden habe, ist dies im Grunde einfach eine Indizierung des Kurses im Zeitverlauf versus dem ersten Kurswert im Diagramm. Also in etwas sowas wie
Benchmark[tn] = Kurs[tn] / Kurs[t0] * 100
(abzüglich 100, damit die Zeitreihe bei 0 beginnt).

Insofern müsste auch ein Gesamtbenchmark für eine Klassifizierung berechenbar sein.
Ungewichtet (für drei enthaltene Wertpapiere, respektive Untergruppen):
GesamtBenchmark = (BenchmarkA * BenchmarkB * BenchmarkC) ^ (1/3)
sprich: dritte Wurzel (da drei Werte) aus dem Produkt der Einzelbenchmarks (wobei diese natürlich noch zu Basis 1 oder Basis 100 sein müssen

Gewichtet (mit den Gewichten 20, 30, 50):
GesamtBenchmark = (BenchmarkA^10 * Benchmark^30 * BenchmarkC^50) ^ (1/100)

**Comment by buchen** [comment]
Ok, jetzt habe ich das verstanden. Irgendwo anders kam auch schon mal die Frage nach synthetischen Benchmarks. Also z.b. 70% DAX und 30% MSCI World.

> Gewichtet (mit den Gewichten 20, 30, 50):

Die Gewichtung sagt ja eigentlich zu wie viel Prozent das Papier der Kategorie zugeordnet ist (man kann die ja aufteilen) und weniger die relative Gewichtung zueinander. Das könnte man aber natürlich durch Unterkategorien erreichen.

Da kann man auf jeden Fall was machen. Ich überlege noch ob man die "Klassifizierungen" nicht ein bisschen überlädt. Versteht man das noch. Oder doch lieber separat anlegen?

**Comment by fuchsst** [comment]
Die Klassifizierungen selbst müssten IMHO ganricht angepasst werden.
Bei Unterkategorien gibt es ja bereits eine Gewichtung, die 1:1 den selben Zweck dient wie hier bei den "synthetischen Benchmarks".

Ein wenig anders verhalten sich die einzelnen Wertpapiere. Aber meiner Ansicht nach nicht grundlegend.
Mal folgendes Gedankenspiel: Wie würde die normale Performance Berechnet werden, wenn ich drei Wertpapiere hätte, die ich alle am selben Tag gekauft habe, und "zufälligerweise" mit jeweils dem selben Wert im Portfolio landen (Preis*Anzahl ist z.B. bei allen drei 1000 €).
Sagen wir nun aber, das ich eine Kategorie habe, in der nur eines der drei Wertpapiere zu 100% einfließt, das zweite meinetwegen nur 50% des Wertes und beim dritten 20% (der Rest ist in irgendeiner anderen Kategorie).
Implizit hätte ich eine Gewichtung von 100/50/20. 
D.h. WP1 fließt zu 59% (100/(100+50+20)) ein, WP2 zu 29% und WP3 zu 12%.

Soll heißen, das bei der Berechnung des Benchmarks auf Basis einzelner Wertpapiere der Prozentsatz zwar nicht direkt der Verteilung entspricht, aber er kann als Verteilungsschlüssel verwendet werden.

**Comment by fuchsst** [comment]
Habe mir das gerade nochmal angesehen.
Vlt. ist es doch Sinnvoller die Gewichtung von der Volumenzuordnung zu trennen.
Auch wenn die Lösung gut und richtig funktionieren würde...Nachteil des oben beschriebenen Gedankens wäre, dass ich keine Gewichtung für einzelne Wertpapiere festlegen kann sondern von der impliziten Gewichtung die sich aus der Aufteilung des WP ergibt abhängig bin.

Eine Idee wäre, die %-Aufteilung eines WP auf verschiedene Kategorien, von der Gewichtung zu trennen. Ersteres dient dann allein dazu die Zusammensetzung des WP (z.b. von Fonds nach Region und Branche) im Tool hinterlegen zu können und funktioniert nur auf WP-Ebene. Die Gewichtung dient dazu Sollwerte vorzugeben (so wie jetzt auch, nur eben zusätzlich auch auf WP-Ebene). 
Um es klar voneinander in der Oberfläche abzugrenzen hätte ich die Spalte "Gewichtung" unter "Klassifizierungen"->"Definition der Klassifizierung" umbenannt (z.B "Wertpapier-Anteil" o.ä.). 
Die Spalte "Gewichtung" unter "Rebalancing" enthält dann für die WP außerdem nicht mehr die %-Aufteilung sonder auch einen Soll-Gewicht, analog wie es schon für die Kategorien hinterlegt werden kann.

**Comment by buchen** [comment]
> Eine Idee wäre, die %-Aufteilung eines WP auf verschiedene Kategorien, von der Gewichtung zu trennen.

Ja, das könnte man machen. Und damit wäre auch die Spalte "Gewichtung" eindeutiger. Auf Ebene der Kategorien muss es auf 100 aufgehen. Auf Ebene der Wertpapiere in einer Kategorie ebenfalls.

Bin mir aber auch nicht sicher wie häufig man so was braucht. Zum Beispiel habe ich einer Kategorie "Emerging Markets" diverse ETFs und so weiter zugeordnet. Als Bench mark will ich hier den MSCI Emerging Markets haben. Mein Benchmark ist ja nicht eine Gruppe von EM ETFs. Was ich damit sagen will, wie häufig ist den eine Klassifizierung gleich dem Benchmark. Auch für Regionen oder Branchen nehme ich doch nicht die (zufällige) Auswahl von Papieren die ich hinterlegt habe, oder? Und wen ich es sowieso separate definiere, dann brauch ich das nicht auch noch in der Klassifizierung machen.

Was ist denn Dein Use Case?

**Comment by fuchsst** [comment]
Genau. Denkbar wäre auch das die Gewichtung generell auf einer Ebene innerhalb der Kategorienhierarchie zu 100 auf geht - also auch wenn innerhalb einer Ebene Kategorien und Wertpapiere gemischt sind (z.B. Regionen: für die großen Länder sind in Europa Unterkategorien angelegt, die kleineren landen als Rest-Summe direkt in Europa).

1) Mein Standardbenchmark den ich mit einblende ist ein thesaurierender ETF auf Basis des MSCI World. Das Problem mit dem ist nur das zwar World drauf steht aber nicht drin ist. Südamerika, Afrika und Asien fehlen hier komplett. In einem "Gesamtbenchmark" würde ich den MSCI World um einen BRIC-Fond ergänzen (irgendwann vlt. auch Afrika dazu).

2) Der zweite Fall ist der Vergleich mit mir selbst. Die Performance wird u.a. auch durch das Timing bestimmt. Ist mein Timing gut  sollte ich trotz vergleichbaren Inhalts oberhalb der Benchmarklinie liegen.

3) Auch der genannte Fall 30% DAX + 70% MSCI World kann durchaus sinnvoll sein. Die 30 DAX-Werte sind eigentlich auch im MSCI World, aber darüber kann ich fokussieren (im Grunde könnte man hier z.B. sagen das die 30 DAX Werte gegenüber den anderen knapp 1600 Werten eine höhere Gewichtung bekommen sollen). 

Mein Use-Case geht hauptsächlich in Richtung 2. (1. würde auch damit abgedeckt, weil mein Portfolio im groben so aufgebaut ist).

**Comment by buchen** [comment]
Nur so nebenbei: Ich ziehe mir auch den MSCI World direkt rein. Erst manuelle die historischen Kurse, dann automatisch mit dieser URL als "Tabelle auf einer Webseite": http://www.onvista.de/index/quote_history.html?ID_NOTATION=3194141&RANGE=3M Ich vergleiche dann den risikobehafteten Teil mit dem Index.

Ich sehe aber den Use Case für solchen "synthetischen Benchmarks". Ich werde nicht ganz schnell dazu kommen, aber ich markieren ihn mal mit einem Milestones - da ist der prominenter. :smile:

**Comment by fuchsst** [comment]
noch eine Idee Anmerkung hierzu: für die Grafiken gibt es ja nun die Option "'Ohne Klassifizierung' einbeziehen". Ich glaube die Option wäre auch innerhalb des Rebalancing hilfreich.
Momentan wird für die SOLL-Gewichte für "Ohne Klassifizierung" "n/a" ausgeweisen, bei den IST-Gewichten aber berücksichtigt. Wenn hier die Möglichkeit bestehen würde die nicht klassifizierten zu ignorieren, wären SOLL- und IST- Gewichte besser vergleichbar.

**Comment by buchen** [comment]
Nicht klassifizierte Werte laufen auch nicht in die SOLL-Gewichtung mit ein. Etwas irritierend ist vielleicht, dass das auf dem ganz obersten Knoten (und nur da) nicht reflektiert wird. Aber hier wollte ich auf jeden Fall die IST Werte des Portfolios anzeigen.

Schlägst Du vor der Kategorie "Ohne Klassifizierung" auch eine SOLL Gewichtung zu geben? Momentan fliessen alle nicht klassifizierten Werte einfach *nicht* in die SOLL Rechnung ein. Wenn wir aber auch den umklassifizierten eine SOLL Gewichtung geben, dann muss man dass auch gewichten. Dann kann aber auch selber eine Kategorie "Rest" anlegen und alle da rein verschieben.

Siehe auch das Beispiel im WIKI https://github.com/buchen/portfolio/wiki/Rebalancing

**Comment by fuchsst** [comment]
Ich glaube wir meinen fast das gleiche.
Das in der obersten Zeile die Gesamtsumme angezeigt wir ist gut so.

Mein Gedanke eher umkehert: auch bei den IST % in der Zeile "Ohne Klassifizierung" "n/a" anzuzeigen.
Beispiel:
ich habe 10.000 Euro davon 6000 in irgendeiner Form auf drei Klassifizierungen verteilt.
Im Soll gebe ich nun mein Wunsch-Gewichte an (z.B. 10%/30%/60%).
Das aktuelle Problem ist nun, das ich bei den IST % nie sinvolle Werte zum vergleichen sehe, da 40% ja schon in der "Ohne Klassifizierung" Zeile landen.

Soll heißen:
SOLL % und SOLL Wert bleiben so wie sie sind
IST % bezieht sich nur auf die Klassifizierungen -> Zusammen 100% (das ist quasi die Änderung; in der Zeile "Ohne Klassifizierung" wird "n/a" angezeigt, und in der obersten Zeile der Klarheit wegen anstelle von 100% auch "n/a"). 
IST Wert bleibt so wie er momentan ist.

Wenn die Option "'Ohne Klassifizierung' einbeziehen" aktiviert ist, ist die Funktionsweise genauso wie jetzt.

**Comment by buchen** [comment]
Ok, jetzt habe ich das Problem verstanden. :smile:  Wenn es unklassifizierte Werte gibt möchte man die IST Prozentzahlen gegen die SOLL Prozentzahlen vergleichen können. Da kann ich was machen.

**Comment by fuchsst** [comment]
ja so hätte man das auch etwas kürzer (und verständlicher) ausdrücken können ;-)

**Comment by MojoDodo** [comment]
Gibt es zu solch einer Funktion schon konkreteres? Das würde die Analysefunktion von PP nochmals enorm aufwerten!
